Mathlib/LinearAlgebra/QuadraticForm/Basis.lean

```diff
@@ -60,4 +62,25 @@ theorem _root_.LinearMap.BilinMap.toQuadraticMap_surjective [Module.Free R M] :
   letI : LinearOrder ι := IsWellOrder.linearOrder WellOrderingRel
   exact ⟨_, toQuadraticMap_toBilin _ b⟩
 
+variable (S) [CommSemiring S] [Algebra S R]
+variable [Module S N] [IsScalarTower S R N]
+
+/-- For each basis `bm`, `toBilinHom` is the map that sends a quadratic map on a module `M` over `R`
+to its associated symmetric bilinear map.  As provided here, this has the structure of an
+`S`-linear map where `S` is a commutative subring of `R`. -/
+noncomputable def toBilinHom (bm : Basis ι R M) : QuadraticMap R M N →ₗ[S] (BilinMap R M N) where
+  toFun Q := Q.toBilin bm
+  map_add' Q₁ Q₂ := by
+    refine bm.ext fun i => bm.ext fun j => ?_
+    obtain h | rfl | h := lt_trichotomy i j
+    · simp [h.ne, h, toBilin_apply, polar_add]
+    · simp [toBilin_apply]
+    · simp [h.ne', h.not_lt, toBilin_apply, polar_add]
+  map_smul' r Q := by
+    refine bm.ext fun i => bm.ext fun j => ?_
+    obtain h | rfl | h := lt_trichotomy i j
+    · simp [h.ne, h, toBilin_apply, polar_smul]
+    · simp [toBilin_apply]
+    · simp [h.ne', h.not_lt, toBilin_apply]
```

Comment:
Can you state these as standalone `simp` theorems first?

Reply:
@eric-wieser done - have I got the names right?